src/odemis/driver/smaract.py

```diff
@@ -2931,6 +2956,16 @@ def _doReference(self, future, axes):
                         logging.warning("Axis %s not referenced after the end of referencing", a)
                         # TODO: Raise some error here
 
+                    # if referenced, move to the safe posiiton (if defined)
+                    if self._pos_deactive_after_ref and self._deactive_pos_after_ref is not None \
+                        and a in self._deactive_pos_after_ref:
+                        deactive_pos = self._deactive_pos_after_ref[a]
+                        logging.info("Moving axis %s to deactivated position %f", a, deactive_pos)
+                        f = self._createMoveFuture()
```

Comment:
It's not a good idea to create a separate Future as it prevents the cancellation from working. I guess it doesn't work to use the same Future only due to the lock. If you convert _moving_lock from Lock() to RLock(), this should be fine.